phpBB/assets/javascript/editor.js

```diff
@@ -24,337 +6,382 @@ function helpline(help) {
+					return m.toUpperCase()
+				}
+			);
+		},
 
-		txtarea.sel_start = sel_start;
-
-		// we ignore the end value for IE, this is already dirty enough and we don't need it
-		caretPos.start = txtarea.sel_start;
-		caretPos.end = txtarea.sel_start;
-	}
-
-	return caretPos;
-}
-
-/**
-* Allow to use tab character when typing code
-* Keep indentation of last line of code when typing code
-*/
-(function($) {
-	$(document).ready(function() {
-		var doc, textarea;
-
-		// find textarea, make sure browser supports necessary functions
-		if (document.forms[form_name]) {
-			doc = document;
-		} else {
-			doc = opener.document;
+		urlencode: function(str){
+			return encodeURIComponent(str);
 		}
 
-		if (!doc.forms[form_name]) {
-			return;
+	},
+
+	/**
+	* Given a name, an object with the attributes and a content, 
+	* this returns a string with the BBCode definition of the content.
+	* NOTE: No escapes are made to the content. All data is used as-is.
+	*
+	* @param name string The name of the BBCode tag
+	* @param attributes object An object with key-value data for the BBCode attributes
+	* @param string content The content of the BBCode tag
+	* @param boolean isSelfClosing Set to true for not return with a closing tag
+	* @return string The BBCode tag given the above parameters
+	*/
+	revertBackToBBCode: function (name, attributes, content, isSelfClosing){
+		var attributeStr = ' ';
+		if(attributes.defaultattr){
+			attributeStr = '="' + attributes.defaultattr + '" ';
+			delete attributes.defaultattr;
 		}
 
-		textarea = doc.forms[form_name].elements[text_name];
+		for(attributeName in attributes){
+			attributeStr += attributeName + '="' + attributes[attributeName] + '" ';
+		}
 
-		phpbb.applyCodeEditor(textarea);
-	});
-})(jQuery);
+		if(attributeStr === ' '){
+			attributeStr = '';
+		}
 
+		return '[' + name + attributeStr + ']' + content + 
+				(isSelfClosing ? '[/' + name + ']' : '');
+	},
+
+
+	/**
+	 * @return A javascript object that allows adding parameters and
+	 * @source Based on s9e\TextFormatter\render.js
+	*/
+	xslt: function (xsl){
+		// older IE has its own way of doing it
+		var standardsBrowser = (typeof DOMParser !== 'undefined' && typeof XSLTProcessor !== 'undefined');
+		if (standardsBrowser) {
+			var xslDoc = (new DOMParser).parseFromString(xsl, 'text/xml');
+
+			var processor = new XSLTProcessor();
+			processor.importStylesheet(xslDoc);
+
+			return {
+				'setParameter': function (name, value){
+					if(!value){
+						if(value === ''){
+							value = ' ';
+						}else{
+							value = '';
+						}
+					}
+					processor.setParameter(null, name, value);
+				},
+
+				'transformToFragment' : function (xml, onDocument){
+					var xmlDoc = (new DOMParser).parseFromString(xml, 'text/xml');
+					// NOTE: importNode() is used because of https://code.google.com/p/chromium/issues/detail?id=266305
+					return onDocument.importNode(processor.transformToFragment(xmlDoc, onDocument), true);
+				}
+			};
+		}else{
```

Comment:
It is proposed, there's a ticket for it, it hasn't been accepted yet. That code predates the time of the proposal. I'll remove it if the proposal gets accepted.